management/commands/aggregate_subject_trials.py

```diff
@@ -315,17 +418,110 @@ def handle(self, *_, **options):
+                return None, [out_file, training_out_file], log_file
+            else:
+                return None, out_file, log_file
 
         # Create dataset
         dset, out_file = self.register_dataset(
-            out_file, file_hash=md5_hash, file_size=file_size, aggregate_hash=aggregate_hash, user=user, revision=revision)
+            out_file, file_hash=md5_hash, file_size=file_size, aggregate_hash=aggregate_hash, user=self.user,
+            revision=self.revision)
 
         # Move log file to new revision folder
         if out_file.parent != log_file.parent:
             log_file = log_file.rename(out_file.with_name(log_file.name))
 
-        return dset, out_file, log_file
+        if compute_training_status:
+            training_dset, training_out_file = self.handle_training_status(trials_table=out_file, rerun=rerun, dry=dry)
+            return [dset, training_dset], [out_file, training_out_file], log_file
+        else:
+            return dset, out_file, log_file
+
+    def handle_training_status(self, trials_table=None, rerun=False, dry=False):
+        """
+        Compute subject training criteria dataset
+
+        Parameters
+        ----------
+        trials_table : pathlib.Path or None
+            Path to the subjectTrials.table dataset on disk, the sessions to use in the training status computation
+            are taken from this file
+        rerun : bool
+            Indicates if the subjectTrials table has been newly created
+        dry: bool
+            Runs aggregate table generation without registration. Output file contains "DRYRUN"
+
+        Returns
+        -------
+        Dataset
+            The Dataset record for the aggregate table.
+        pathlib.Path
+            The output file path.
+        """
+
+        # Try except as there are cases where it fails, and we don't want that to stop the trials table registration
+        try:
+            qs = Dataset.objects.filter(
+                name='_ibl_subjectTraining.table.pqt', object_id=self.subject.id, default_dataset=True)
+
+            # If a dataset already exists and rerun is false, we do not do anything
+            if qs.count() > 0 and not rerun:
+                logger.info(f'Training status file exists and rerun is {rerun}; exiting')
+                return None, None
+
+            # Otherwise either the dataset doesn't exist or needs to be recomputed
+            # If the trials table file hasn't just been created we need to find the relevant file
+            if trials_table is None:
+                dset = Dataset.objects.get(name='_ibl_subjectTrials.table.pqt', object_id=self.subject.id,
+                                           default_dataset=True)
+                trials_table = next(Path('/mnt/ibl/aggregates').joinpath(dset.file_records.all()[0].relative_path).
+                                    parent.glob(f'_ibl_subjectTrials.table.{str(dset.id)}.pqt'))
```

Comment:
No need to assert after a glob, but this could be simplified with `self.output_path / alfiles.add_uuid_string(dset.file_records.first().relative_path), dset.pk)` (in which case the assert is necessary)